server/room.ts

```diff
@@ -45,193 +48,329 @@ export class Room {
-        this.roster.forEach((user, i) => {
-          if (user.id === socket.id) {
-            this.roster[i].isController = true;
-          } else {
-            this.roster[i].isController = false;
-          }
-        });
-        this.cmdHost(
-          undefined,
-          'vbrowser://' + this.vBrowser.pass + '@' + this.vBrowser.host
-        );
-        io.of(roomId).emit('roster', this.roster);
-      });
-      socket.on('CMD:stopVBrowser', () => this.resetRoomVM());
-      socket.on('CMD:changeController', (data: string) => {
-        this.roster.forEach((user, i) => {
-          if (user.id === data) {
-            this.roster[i].isController = true;
-          } else {
-            this.roster[i].isController = false;
-          }
-        });
-        io.of(roomId).emit('roster', this.roster);
-      });
-      socket.on('CMD:askHost', () => {
-        socket.emit('REC:host', this.getHostState());
-      });
-      socket.on('signal', (data: { to: string; msg: string }) => {
-        io.of(roomId)
-          .to(data.to)
-          .emit('signal', { from: socket.id, msg: data.msg });
-      });
-      socket.on(
-        'signalSS',
-        (data: { to: string; sharer: boolean; msg: string }) => {
-          io.of(roomId).to(data.to).emit('signalSS', {
-            from: socket.id,
-            sharer: data.sharer,
-            msg: data.msg,
-          });
-        }
-      );
-
-      socket.on('disconnect', () => {
-        let index = this.roster.findIndex((user) => user.id === socket.id);
-        const removed = this.roster.splice(index, 1)[0];
-        io.of(roomId).emit('roster', this.roster);
-        if (removed.isScreenShare) {
-          // Reset the room state since we lost the screen sharer
-          this.cmdHost(socket, '');
-        }
-        delete this.tsMap[socket.id];
-        // delete nameMap[socket.id];
-      });
+      socket.on('CMD:name', this.setName);
+      socket.on('CMD:picture', this.addPicture);
+      socket.on('CMD:host', this.host);
+      socket.on('CMD:addVideoToPlaylist', this.addVideoToPlaylist);
+      socket.on('CMD:play', this.playVideo);
+      socket.on('CMD:pause', this.pauseVideo);
+      socket.on('CMD:seek', this.seekVideo);
+      socket.on('CMD:ts', this.sendVideoTimestamp);
+      socket.on('CMD:chat', this.sendChatMessage);
+      socket.on('CMD:joinVideo', this.joinVideo);
+      socket.on('CMD:leaveVideo', this.leaveVideo);
+      socket.on('CMD:joinScreenShare', this.joinScreenShare);
+      socket.on('CMD:leaveScreenShare', this.leaveScreenShare);
+      socket.on('CMD:startVBrowser', this.startVBrowser);
+      socket.on('CMD:stopVBrowser', this.stopVBrowser);
+      socket.on('CMD:changeController', this.changeController);
+      socket.on('CMD:askHost', this.askHost);
+      socket.on('signal', this.emitSignal);
+      socket.on('signalSS', this.emitSSSignal);
+      socket.on('disconnect', this.disconnect);
     });
   }
 
+  setName = (data: string) => {
```

Comment:
I think all of these need to be passed the socket context, or else they'll all be sent to the same client (the last connection to the server).

At least, that's what it looks like without running it, you can test by opening a second tab and seeing if stuff is routed properly.

Reply:
Let me test that! :)